rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,633 @@
+        For example, if table a.b.t exists, using 'SELECT NAMESPACE IN a' this would translate into
+        `GET /namespaces?parent=a` and must return Namepace.of("a","b").
+      operationId: listNamespaces
+      parameters:
+        - name: parent
+          in: query
+          description: Optional parent namespace under which to list namespaces. When empty, list top-level namespaces.
+          required: false
+          schema:
+            type: string
+          example: ?parent=accounting.tax
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found (Parent namespace does not exist)
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        '409':
+          $ref: '#/components/responses/NamespaceAlreadyExistsResponse'
+
+  /namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '404':
+          $ref: '#/components/responses/NoSuchNamespaceResponse'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description: Check if a namespace exists.
+      responses:
+        '200':
+          description: OK - Namesapce exists
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergResponseObject'
+              example: { "data": { "dropped": true } }
```

Comment:
I believe, from other discussion (much of which is lost in this PR becoming rather large), that the idea was to return a response body where possible to be able to more easily add additional metadata without having to completely change the response (e.g. the response type will still be an object).

So I don't disagree that it can be gleaned from the response status code, but the overall consensus so far was to return a body where possible so that the response body can be updated in the future and not require fully changing the type of the response (e.g. even if this just returned `{}`).

Reply:
Sorry about reopening something discussed previously. I guess that's one of the shortcomings of Github reviews which hide previous conversations?

That said, I don't necessarily see this as incompatible as a successful operation can still return a body? But what I would to be valuable is that the HTTP status code would actually give a clear indication if the operation was successful or not, similar to other operations (like when `404` is returned when the namespace or the table doesn't exist)?

I guess maybe my grief is with the iceberg API and the use of a boolean return value for `Catalog#dropTable(TableIdentifier)` to notify of an error condition whereas `Catalog#createTable(TableIdentifier, Schema)` would throw an exception? Using a return value could cause the error to go unnoticed if the caller is not checking the value.

Reply:
> I guess maybe my grief is with the iceberg API and the use of a boolean return value for Catalog#dropTable(TableIdentifier) to notify of an error condition whereas Catalog#createTable(TableIdentifier, Schema) would throw an exception? Using a return value could cause the error to go unnoticed if the caller is not checking the value.

That's fair. So this is sort of just a sample body. We could use a 4xx here on failure, to ensure that it's not missed. And then the REST catalog could translate the correct 4xx into the drop failing.

For things that are majorly issues / concerns with the Iceberg API in general, I'd suggest possibly opening a separate issue for that. The focus here is just to be able to get started on some sort of generalized enough HTTP API so that people who didn't want to write their own catalog could plug into this one (which has come up a few times from various people).

For this specific one, the 200 is really what indicates the drop was successful. The body is just additional metadata really (so that it could be evolved without having to change from not parsing a body to parsing a body.

That said, I agree that some sort of 4xx should be thrown if the purge was requested and didn't work. What would you suggest as far as 4xx responses go?